django/db/models/fields/__init__.py

```diff
@@ -944,7 +944,12 @@ def get_prep_value(self, value):
         value = super(AutoField, self).get_prep_value(value)
         if value is None:
             return None
-        return int(value)
+        try:
+            return int(value)
+        except (TypeError, ValueError):
+            raise TypeError("{}: expected a number, not {}".format(
```

Comment:
Use `raise .. from ...` as in 6e55e1d88a5c4453e25f0caf7ffb68973de5c0ba to get the benefit of exception chaining.
Perhaps the original exception type should be preserved rather than converting ValueError to TypeError, otherwise that change could be backwards-incompatible (although it might be more helpful in the long run if user code only has to catch TypeError).
For the message, perhaps something along the lines of `"Field '{}' expected a number but got {!r}"` would be more clear.
Some test updates are required.